dbt/adapters/databricks/connections.py

```diff
@@ -98,7 +98,12 @@ def unique_field(self) -> str:
         return self.host
 
     def _connection_keys(self) -> Tuple[str, ...]:
-        return "host", "http_path", "database", "schema", "session_properties"
+        connection_keys = ["host", "http_path", "schema"]
+        if self.database:
+            connection_keys.insert(2, "catalog")
```

Comment:
Not sure if the order matters but if not we can just append it after "schema".

Reply:
I just felt it weird that `catalog` comes after `schema` because `catalog` is a larger concept that `schema`.